usb_flash_drive_copy.py

```diff
@@ -0,0 +1,543 @@
+
+    @staticmethod
+    def get_copy_info_csv(directory: Path) -> Path:
+        """Get the location of the copy information CSV file of this OTCamera."""
+        return Path(directory, f"{get_hostname()}{COPY_INFO_CSV_SUFFIX}")
+
+    @staticmethod
+    def create_new(src_dir: Path, dest_dir: Path, filetype: str):
+        dest_dir.mkdir(parents=True, exist_ok=True)
+        copy_csv_file = CopyInformation.get_copy_info_csv(dest_dir)
+        copy_csv_file.touch()
+
+        video_filepaths = get_video_files(src_dir, filetype)
+        videos: set[Video] = set()
+
+        for video_filepath in video_filepaths:
+            video = Video(video_filepath.name, video_filepath, False, False)
+            videos.add(video)
+
+        return CopyInformation(videos, copy_csv_file, src_dir, dest_dir)
+
+    def to_dict(self) -> list[dict]:
+        dict_videos: list[dict] = []
+        for video in self.get_sorted_videos():
+            video_dict = video.to_dict()
+            dict_videos.append(video_dict)
+        return dict_videos
+
+
+@dataclass
+class UsbFlashDrive:
+    """Wrapper to usb flash drives providing mount and unmount function.
+
+    IMPORTANT: Requires an entry in /etc/fstab to give user permission to access
+    usb mount device.
+
+    `self.mount_point` must be the same as in /etc/fstab.
+
+    Raises:
+        UsbFlashDriveNotMountableError: If not able to mount usb flash drive.
+    """
+
+    mount_point: Path
+
+    def mount(self) -> None:
+        """Mount USB flash drive.
+
+        IMPORTANT: Requires an entry in /etc/fstab to give user permission to access
+        usb mount device.
+
+        `self.mount_point` must be the same as in /etc/fstab.
+        """
+        if self.mount_point.is_mount():
+            log.write("USB flash drive already mounted", log.LogLevel.WARNING)
+            return
+
+        self.mount_point.mkdir(parents=True, exist_ok=True)
+
+        return_code: int = subprocess.call(f"sudo mount {self.mount_point}", shell=True)
+        if return_code != 0:
+            raise UsbFlashDriveNotMountableError(
+                (f"Unable to mount USB flash drive to '{self.mount_point}'!",)
+            )
+        log.write("USB flash drive mounted.")
+
+    def unmount(self) -> None:
+        """Unmount USB flash drive."""
+        if not self.mount_point.is_mount():
+            log.write("USB flash drive already unmounted", log.LogLevel.WARNING)
+            return
+
+        return_code: int = subprocess.call(
+            f"sudo umount {self.mount_point}", shell=True
+        )
+        if return_code != 0:
+            raise UsbFlashDriveUnmountableError(
+                f"Unable to unmount USB flash drive from '{self.mount_point}'!"
+            )
+        log.write("USB flash drive unmounted.")
+
+
+class OTCameraUsbCopier(Observer):
+    def __init__(
+        self,
+        power_led: Led,
+        wifi_led: Led,
+        rec_led: Led,
```

Comment:
I thought using the same names of the leds that are currently used in the context of OTCamera would make it easier to understand which physical LEDs on the OTCamera are talked about?
Maybe it would be better to name them:

```python
    def __init__(
        self,
        otcamera_power_led: Led,
        otcamera_wifi_led: Led,
        otcamera_rec_led: Led,
    ):
    # init code
```
What do you think would be best?

Reply:
After reading the rest of the code, I assumed that the led names are identical to the labels on the case. In this case, let the variable names as they are. We can discuss this with @frunika some time in the future.
I think `status` would fit well but the led should also be named `status` on the case.